src/hooks/safe-apps/usePinnedSafeApps.ts

```diff
@@ -0,0 +1,23 @@
+import { useState, useEffect } from 'react'
+import local from '@/services/local-storage/local'
+
+type ReturnType = {
+  pinnedSafeAppIds: Set<number>
+  updatePinnedSafeApps: (newPinnedSafeAppIds: Set<number>) => void
+}
+
+const pinnedSafeAppsIdsKey = 'pinnedSafeAppsIds'
+
+const usePinnedSafeApps = (): ReturnType => {
+  const [pinnedSafeAppIds, updatePinnedSafeApps] = useState<Set<number>>(
+    () => new Set(local.getItem<number[]>(pinnedSafeAppsIdsKey) || []),
+  )
+
+  useEffect(() => {
+    local.setItem(pinnedSafeAppsIdsKey, Array.from(pinnedSafeAppIds))
+  }, [pinnedSafeAppIds])
+
+  return { pinnedSafeAppIds, updatePinnedSafeApps }
+}
```

Comment:
You can probably use `useLocalStorage` here instead.

Reply:
It doesn't work with data types that aren't part of json. I need to use a set here because while rendering, we check if the app is pinned and the set has an `O(1)` lookup.

Reply:
But you're deserializing a local item from `number[]` into a set anyway? You can do the same with `useLocalStorage`.

Reply:
This is a version with `useLocalStorage`:

```
const usePinnedSafeAppsHook = (): ReturnType => {
  const [pinnedSafeAppIdsStorage, updatePinnedSafeAppsStorage] = useLocalStorage<number[]>(pinnedSafeAppsIdsKey, [])

  const pinnedSafeAppIds = useMemo(() => new Set(pinnedSafeAppIdsStorage), [pinnedSafeAppIdsStorage])
  const updatePinnedSafeApps = useCallback(
    (newPinnedSafeAppIds: Set<number>) => {
      updatePinnedSafeAppsStorage(Array.from(newPinnedSafeAppIds))
    },
    [updatePinnedSafeAppsStorage],
  )

  return { pinnedSafeAppIds, updatePinnedSafeApps }
}
```

Not sure if it wins us anything, the amount of LOC is the same, with the only exception that we sneaked in one more level of abstraction (`hook` -> `useLocalStorage` -> `Storage` service). I think `useLocalStorage` is a breeze when working with serializable types, but when working with something non-serializable, it's not super handy

Reply:
I see, thanks for taking the time to explain. Resolving.